test/integration/targets/ansible-galaxy-collection/tasks/list.yml

```diff
@@ -31,6 +31,28 @@
       - "'dev.collection2 placeholder' in list_result.stdout"
       - "'dev.collection3 *' in list_result.stdout"
 
+- name: list collections in yaml format
+  command: ansible-galaxy collection list -o yaml
```

Comment:
I'd expect extra tests for listing specific collections too. Like `ansible-galaxy collection list -o yaml ns1.coll2` and `ansible-galaxy collection list -o yaml ns1.coll2:1.0` maybe...

Reply:
Added below